### PR TITLE
Feature - allow ResourceManager getPath to return a relative path based on a known base

### PR DESCRIPTION
**Pull request by rossriley**
As discussed in #2140

### CODE REVIEW

File: src/Configuration/ResourceManager.php [hunk 1]
```diff
@@ -148,6 +152,27 @@ public function getPath($name)
...
+    
+    /**
+     * Takes a known path with relative additional atoms and returns a new path
+     * for instance constructRelativePath('public/images/example')
+     *
+     * @return AbsoluteUnixPath Object
+     **/
+    public function constructRelativePath($name)
+    {
+        list($firstAtom) = explode('/', $name);
```
Comment (GawainLynch) [inline src/Configuration/ResourceManager.php]: Isn't this assuming *NIX?

Comment (rossriley) [inline src/Configuration/ResourceManager.php]: This is only an access interface, if you ask for getPath("cache/test") you will still get back an OS specific path, just we're using forward slashes as a divider, the same as we do in config->get('general/branding')

**Comment by CarsonF** [comment]
I just did something similar, #2601. 
```php
$app['resources']->getPath('root', true)->joinAtoms('foo', 'bar');
```

Less code perhaps?

**Comment by rossriley** [comment]
constructRelativePath refers to the fact that it is relative to the base that you pass in. The interface that you use is:
    $app['resources']->getPath('root/vendor');
    $app['resources']->getPath('public/files/demos');

It means that we can get rid of a lot of instances across the codebase where you see stuff like this:

    $app['resources']->getPath('rootpath') . '/src';

**Comment by CarsonF** [comment]
Right, I was going for the same thing. 
Ironic that we both had this thought at the same time. 

Now that I look at this more, I like :+1: 

Can we still have optional argument from my PR so that other PathInterface methods can be used?

**Comment by bobdenotter** [comment]
Looks good to me! :100:

**Pull request merged**